pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2437,3 +2449,59 @@ func TestServiceRegistryCreateWithoutExistingNetwork(t *testing.T) {
 		t.Fatalf("expected error; got none")
 	}
 }
+
+func TestServiceRegistryCreateWithNetworkOfArktosIPAM(t *testing.T) {
+	origFeatureGate := newFeatureGateWithPerNetworkServiceIPAlloc()
+	defer func() {
+		utilfeature.DefaultFeatureGate = origFeatureGate
+	}()
+
+	storage, registry, server := NewTestREST(t, nil)
+	storage.networks = &dummyNetworkGetter{ipamType: "Arktos", cidr: "3.4.5.0/24"}
+	_, ipRange, _ := net.ParseCIDR("3.4.5.0/24")
+	defer server.Terminate(t)
+
+	svc := &api.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Tenant:    "system",
+			Namespace: "default",
+		},
+		Spec: api.ServiceSpec{
+			Selector:        map[string]string{"bar": "baz"},
+			SessionAffinity: api.ServiceAffinityNone,
+			Type:            api.ServiceTypeClusterIP,
+			Ports: []api.ServicePort{{
+				Port:       6502,
+				Protocol:   api.ProtocolTCP,
+				TargetPort: intstr.FromInt(6502),
+			}},
+		},
+	}
+	ctx := genericapirequest.NewDefaultContext()
+	created_svc, err := storage.Create(ctx, svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	created_service := created_svc.(*api.Service)
+
+	ip := net.ParseIP(created_service.Spec.ClusterIP)
+	if !ipRange.Contains(ip) {
+		t.Errorf("expected ip in %s; got %s", ipRange, created_service.Spec.ClusterIP)
+	}
+
+	srv, err := registry.GetService(ctx, svc.Name, &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if srv == nil {
+		t.Errorf("Failed to find service: %s", svc.Name)
```

Comment:
should be Fatalf here. If srv==nil, net.ParseIP(srv.Spec.ClusterIP) is doomed to run into failure.

Reply:
will change to Fatalf as suggested